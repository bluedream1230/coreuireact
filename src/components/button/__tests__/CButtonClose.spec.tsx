import * as React from 'react'
import { render /* ,screen */ /*,fireEvent */ /* ,waitFor */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CButtonClose } from '../../../index'

test('loads and displays CButtonClose component', async () => {
  const { container } = render(<CButtonClose>Test</CButtonClose>)
  expect(container).toMatchSnapshot()
})

test('CButtonClose customize', async () => {
  const { container } = render(
    <CButtonClose white={true} disabled={true} className="bazinga">
      Test
    </CButtonClose>,
  )
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('bazinga')
  expect(container.firstChild).toHaveClass('btn')
  expect(container.firstChild).toHaveClass('btn-close')
  expect(container.firstChild).toHaveClass('btn-close-white')
  expect(container.firstChild).toHaveAttribute('disabled')
})
