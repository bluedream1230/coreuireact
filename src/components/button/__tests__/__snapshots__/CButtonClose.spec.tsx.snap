// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CButtonClose customize 1`] = `
<div>
  <button
    aria-label="Close"
    class="btn btn-close btn-close-white bazinga"
    disabled=""
  >
    Test
  </button>
</div>
`;

exports[`loads and displays CButtonClose component 1`] = `
<div>
  <button
    aria-label="Close"
    class="btn btn-close"
  >
    Test
  </button>
</div>
`;
